api/handlers.go

```diff
@@ -0,0 +1,73 @@
+package api
+
+import (
+	"encoding/csv"
+	"log"
+	"net/http"
+	"os"
+	"strconv"
+
+	"github.com/gorilla/mux"
+	"github.com/linhnguyen124/ondemand-go-bootcamp/model"
+)
+
+func GetPokemonByID(w http.ResponseWriter, r *http.Request) {
+	vars := mux.Vars(r)
+	idStr := vars["id"]
+
+	id, err := strconv.Atoi(idStr)
+	if err != nil {
+		http.Error(w, "Invalid ID", http.StatusBadRequest)
+		return
+	}
+
+	pokemonList, err := readPokemonCSV("./resources/pokemon.csv")
+	if err != nil {
+		log.Println("Error reading CSV:", err)
+		http.Error(w, "Internal Server Error", http.StatusInternalServerError)
+		return
+	}
+
+	for _, pokemon := range pokemonList {
+		if pokemon.ID == id {
+			w.WriteHeader(http.StatusOK)
+			w.Write([]byte(pokemon.Name))
+			return
+		}
+	}
+
+	http.Error(w, "Pokemon not found", http.StatusNotFound)
+}
+
+func readPokemonCSV(filePath string) ([]*model.Pokemon, error) {
+	//Read the CSV file and return the data as a slice of Pokemon structs
+
+	file, err := os.Open(filePath)
+	if err != nil {
+		return nil, err
+	}
+	defer file.Close()
+
+	reader := csv.NewReader(file)
+	reader.FieldsPerRecord = -1
+	records, err := reader.ReadAll()
+	if err != nil {
+		return nil, err
+	}
+
+	pokemonList := []*model.Pokemon{}
```

Comment:
I think using a pointer here may not be the best idea, as you are iterating over the information and if there was a error you would skip that record, you could use just []model.Pokemon{} This implies a change in the methods to use []model.Pokemon without the * as it could cost more i/o operations, remember that a pointer is the only way to keep the same object otherwise it will be a copy, in this case, it would be better to not use a pointer.

Reply:
Good idea I see the change not using pointer can avoids potential additional memory allocations. Thank for your suggestion!